spec/rubocop/cop/style/redundant_freeze_spec.rb

```diff
@@ -23,6 +23,11 @@
   it_behaves_like 'immutable objects', '1.5'
   it_behaves_like 'immutable objects', ':sym'
   it_behaves_like 'immutable objects', ':""'
+  it_behaves_like 'immutable objects', "ENV['foo']"
+  it_behaves_like 'immutable objects', "'foo'.count"
+  it_behaves_like 'immutable objects', '(1 + 2)'
```

Comment:
I guess we also need tests which show what happens for objects other than numbers.

Reply:
I added tests for comparing strings, comparing variables, adding string, and adding variables.